src/test/java/org/jabref/logic/integrity/ValidCitationKeyCheckerTest.java

```diff
@@ -0,0 +1,30 @@
+package org.jabref.logic.integrity;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.jabref.logic.l10n.Localization;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ValidCitationKeyCheckerTest {
+
+    private final ValidCitationKeyChecker checker = new ValidCitationKeyChecker();
+
+    @ParameterizedTest
+    @MethodSource("provideCitationKeys")
+    void citationKeyValidity(Optional optionalArgument, String citationKey, String errorMessage) {
+        assertEquals(optionalArgument, checker.checkValue(citationKey), errorMessage);
+    }
+
+    private static Stream<Arguments> provideCitationKeys() {
+        return Stream.of(
+                Arguments.of(Optional.of(Localization.lang("empty citation key")), "", "Citation key not empty"),
+                Arguments.of(Optional.empty(), "Seaver2019", "Invalid citation key"),
+                Arguments.of(Optional.of(Localization.lang("Invalid citation key")), "Seaver_2019}", "Valid citation key"));
+    }
```

Comment:
Please do not output error details at assertion statements. Better add Java comments above if it is complex. Otherwise, just leave it out. This is a) in line with other existing tests, b) mostly, the text can be "trivially" derived from the test case itself. 

```suggestion
    void citationKeyValidity(Optional optionalArgument, String citationKey, String errorMessage) {
        assertEquals(optionalArgument, checker.checkValue(citationKey));
    }

    private static Stream<Arguments> provideCitationKeys() {
        return Stream.of(
                Arguments.of(Optional.of(Localization.lang("empty citation key")), ""),
                Arguments.of(Optional.empty(), "Seaver2019"),
                Arguments.of(Optional.of(Localization.lang("Invalid citation key"))}", "Valid citation key"));
    }
```

Reply:
Alright, thanks for the explanation. Just commited your suggestion.